gi_wrapper_cpp/dicom_reader.cpp

```diff
@@ -68,35 +78,48 @@ double* DicomReader::acquireImage(DataSet* dataSet, int& dims,
         exit(-1);
     }
 
-
-    double rescaleSlope = 1.0;
-    double rescaleIntercept = 0.0;
-    if (dataSet->bufferExists(TagId(tagId_t::DoseGridScaling_3004_000E), 0))
-        rescaleSlope = dataSet->getDouble(TagId(tagId_t::DoseGridScaling_3004_000E), 0);
-    else
-    {
-        if (dataSet->bufferExists(TagId(tagId_t::RescaleSlope_0028_1053), 0))
-            rescaleSlope = dataSet->getDouble(TagId(tagId_t::RescaleSlope_0028_1053), 0);
-
-        if (dataSet->bufferExists(TagId(tagId_t::RescaleIntercept_0028_1052), 0))
-            rescaleIntercept = dataSet->getDouble(TagId(tagId_t::RescaleIntercept_0028_1052), 0);
-    }
+    double rescaleSlope = im.GetSlope();
+    double rescaleIntercept = im.GetIntercept();
 
     double* array = new double[xNumber * yNumber * zNumber];
```

Comment:
This explicitly allocated array exits function scope - this may be a memory leak if somebody forgets to deallocate that when using this function - please change the signature of the function to return std::vector<double> instead of some double pointer